queries/vhdl/highlights.scm

```diff
@@ -0,0 +1,407 @@
+  ; "cover"
+  "disconnect"
+  "entity"
+  ; "fairness"
+  "file"
+  "force"
+  "generate"
+  "generic"
+  "group"
+  "label"
+  "literal"
+  "map"
+  "new"
+  "package"
+  "parameter"
+  "port"
+  "property"
+  "range"
+  "reject"
+  "release"
+  ; "restrict"
+  "sequence"
+  "transport"
+  "unaffected"
+  "view"
+  ; "vmode"
+  ; "vpkg"
+  ; "vprop"
+  "vunit"
+] @keyword
+
+[
+  (ALL)
+  (OTHERS)
+  "<>"
+  (DEFAULT)
+  (OPEN)
+] @constant.builtin
+
+[
+  "is"
+  "begin"
+  "end"
+] @keyword
+
+(parameter_specification
+  "in" @keyword)
+
+[
+  "process"
+  "wait"
+  "on"
+  "until"
+] @keyword.coroutine
+
+(timeout_clause
+  "for" @keyword.coroutine)
+
+[
+  "function"
+  "procedure"
+] @keyword.function
+
+[
+  "to"
+  "downto"
+  "of"
+] @keyword.operator
+
+[
+  "library"
+  "use"
+] @keyword.import
+
+[
+  "subtype"
+  "type"
+  "record"
+  "units"
+  "constant"
+  "signal"
+  "variable"
+] @keyword.type
+
+[
+  "protected"
+  "private"
+  "pure"
+  "impure"
+  "inertial"
+  "postponed"
+  ; "strong"
+  "guarded"
+  "out"
+  "inout"
+  "linkage"
+  "buffer"
+  "register"
+  "bus"
+  "shared"
+] @keyword.modifier
+
+(mode
+  "in" @keyword.modifier)
+
+(force_mode
+  "in" @keyword.modifier)
+
+[
+  "while"
+  "loop"
+  "next"
+  "exit"
+] @keyword.repeat
+
+(for_loop
+  "for" @keyword.repeat)
+
+(block_configuration
+  "for" @keyword)
+
+(configuration_specification
+  "for" @keyword)
+
+(component_configuration
+  "for" @keyword)
+
+(end_for
+  "for" @keyword)
+
+"return" @keyword.return
+
+[
+  "assert"
+  "report"
+  "severity"
+] @keyword.debug
+
+[
+  "if"
+  "then"
+  "elsif"
+  "case"
+] @keyword.conditional
+
+(when_element
+  "when" @keyword.conditional)
+
+(case_generate_alternative
+  "when" @keyword.conditional)
+
+(else_statement
+  "else" @keyword.conditional)
+
+(else_generate
+  "else" @keyword.conditional)
+
+[
+  "with"
+  "select"
+] @keyword.conditional.ternary
+
+(when_expression
+  "when" @keyword.conditional.ternary)
+
+(else_expression
+  "else" @keyword.conditional.ternary)
+
+(else_waveform
+  "else" @keyword.conditional.ternary)
+
+(else_expression_or_unaffected
+  "else" @keyword.conditional.ternary)
+
+"null" @constant.builtin
+
+(user_directive) @keyword.directive
+
+(protect_directive) @keyword.directive
+
+(warning_directive) @keyword.directive
+
+(error_directive) @keyword.directive
+
+(if_conditional_analysis
+  "if" @keyword.directive)
+
+(if_conditional_analysis
+  "then" @keyword.directive)
+
+(elsif_conditional_analysis
+  "elsif" @keyword.directive)
+
+(else_conditional_analysis
+  "else" @keyword.directive)
+
+(end_conditional_analysis
+  "end" @keyword.directive)
+
+(end_conditional_analysis
+  "if" @keyword.directive)
+
+(directive_body) @keyword.directive
+
+(directive_constant_builtin) @constant.macro
+
+(directive_error) @comment.error
+
+(directive_protect) @keyword.directive
+
+(directive_warning) @comment.warning
+
+[
+  (condition_conversion)
+  (unary_operator)
+  (logical_operator)
```

Comment:
I think `(logical_operator)` is more suited in `@keyword.operator`

Reply:
True - I'll move it (and others like it).

What about `multiplying_operator`?  It contains `*`, `/`, `mod` and `rem`.  Should I split those into `operator` and `keyword.operator`?

Reply:
The parser should just expose `"*"` and `"/"` as anonymous nodes, and `mod` and `rem` as choices of `multiplying_operator`

Reply:
My question is more philosophical...
I'm leaning towards "yes", simply for consistency.  All symbol-style operators as `operator` and all letter-style operators as `keyword.operator`, regardless of their meaning.

Reply:
Rule of thumb: if it's a symbol, it's an `@operator`; if it's a word, it's a `@keyword.operator`.